data/src/languages_update.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import collections
+import json
+import operator
+import os
+import sys
+
+import unicodedataplus
+from data.src.codes import LANGUAGES_PATH, TSV_DIRECTORY_PATH
+
+from typing import Dict, Tuple, Union
+
+def _detect_best_script_name(
+    word: str, strict: bool = True
+) -> Union[Tuple[str, float], None]:
+    """Returns the most likely script name (rather than ISO 15924 code) the
+    word belongs to along with the corresponding confidence expressed as a
+    maximum likelihood estimate computed over the `word` sample. If `strict`
+    is enabled, then all the characters must belong to the same script and
+    `(None, None)` is returned on failure.
+
+    Example: "ژۇرنال" -> ("Arabic", 1.0).
+    """
+    script_counts: Dict[str, float] = collections.defaultdict(float)
+    for char in word:
+        script_counts[unicodedataplus.script(char)] += 1.0
+    script_probs = [(s, script_counts[s] / len(word)) for s in script_counts]
+    script_probs.sort(key=operator.itemgetter(1), reverse=True)
+    if strict and len(script_probs) != 1:
+        return None
+    else:
+        # The script names in Unicode data tables have underscores instead of
+        # whitespace to enable parsing. See:
+        #   https://www.unicode.org/Public/13.0.0/ucd/Scripts.txt
+        return script_probs[0][0], script_probs[0][1]
+
+
+def _remove_duplicates(script_dict: Dict[str, str]) -> Dict[str, str]:
+    '''
```

Comment:
Please use `"""` rather than `'''` throughout, and also `"`" rather than `'` except if there's some good reason to prefer the other.